packages/blade/src/components/ActionList/ActionListItem.tsx

```diff
@@ -0,0 +1,240 @@
+import React from 'react';
+import styled from 'styled-components';
+import { StyledActionListItem } from './StyledActionListItem';
+import { componentIds } from './componentIds';
+import type { StyledActionListItemProps } from './getBaseActionListItemStyles';
+import Box from '~components/Box';
+import type { IconComponent } from '~components/Icons';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import { Text } from '~components/Typography';
+import { isReactNative, makeAccessible, makeSize } from '~utils';
+import { Checkbox } from '~components/Checkbox';
+
+const ActionListItemContext = React.createContext<{
+  intent?: ActionListItemProps['intent'];
+}>({});
+
+const SectionDivider = styled(Box)((props) => ({
+  // @TODO: replace this with token value if we add 1px token
+  height: makeSize(1),
+  backgroundColor: props.theme.colors.surface.border.normal.lowContrast,
+  margin: `${makeSize(props.theme.spacing[1])} ${makeSize(props.theme.spacing[3])}`,
+}));
+
+const StyledActionListSectionTitle = styled(Box)((props) => ({
+  padding: makeSize(props.theme.spacing[3]),
+}));
+
+const ActionListSection = ({
+  title,
+  children,
+  hideDivider,
+}: {
+  title: string;
+  children: React.ReactNode[] | React.ReactNode;
+  hideDivider?: boolean;
+}): JSX.Element => {
+  return (
+    <Box>
+      <StyledActionListSectionTitle>
+        <Text color="surface.text.muted.lowContrast" size="small" weight="bold">
+          {title}
```

Comment:
I added `role=group` here so that should announce the title. It does have a bit strange behaviour in how it counts the options but I checked few dropdowns from github and they seem to announce it in the similar way so it is consistent with that.

For checking the title, I think we can rely on TS. If the designer has given them a section with title, then that is the only way to add that title in code so don't think so anyone will miss it. Even if they do, it will look strange and will be caught in reviews.

Just wanted to avoid runtime checks as much as possible because we already have too many to make the dropdown work.